src/main/java/com/google/sps/data/Activity.java

```diff
@@ -1,20 +1,37 @@
 package com.google.sps.data;
 
+import com.google.appengine.api.datastore.Key;
+
 public class Activity {
+  private Key id;
   private String name;
   private Category category;
   private String url;
 
+
   public static enum Category {
     GAMES, VIDEOS, ARTICLES;
   }
 
-  public Activity(String name, Category category, String url) {
+  public Activity(Key id, String name, Category category, String url) {
+    this.id = id;
     this.name = name;
     this.category = category;
     this.url = url;
   }
 
+  public Key getId() {
+    return id;
+  }
+  
+  public String getName() {
+    return name;
+  }
+
+  public String getUrl() {
+    return url;
+  }
+
```

Comment:
add new line at EOF

Reply:
There is a new line.

Reply:
You would see a red mark at the end of the file on github if there wasn't.

Reply:
That's what I always thought too. But when I view the file, I don't see the end of line. :( BUt good, if it's there then you're good.